README.md

```diff
@@ -40,99 +41,159 @@ This package also provides optimized functions to compute column-wise and pairwi
-Here, `dist` is an instance of a distance type: for example, the type for Euclidean distance is `Euclidean` (more distance types will be introduced in the next section). You can compute the Euclidean distance between `x` and `y` as
+Here, `dist` is an instance of a distance type: for example, the type for Euclidean
+distance is `Euclidean` (more distance types will be introduced in the next section).
+You can compute the Euclidean distance between `x` and `y` as
 
 ```julia
 r = evaluate(Euclidean(), x, y)
 r = Euclidean()(x, y)
 ```
 
-Common distances also come with convenient functions for distance evaluation. For example, you may also compute Euclidean distance between two vectors as below
+Common distances also come with convenient functions for distance evaluation. For
+example, you may also compute Euclidean distance between two vectors as below
 
 ```julia
 r = euclidean(x, y)
 ```
 
-#### Computing distances between corresponding columns
+### Computing distances between corresponding objects ("column-wise")
 
-Suppose you have two `m-by-n` matrix `X` and `Y`, then you can compute all distances between corresponding columns of `X` and `Y` in one batch, using the `colwise` function, as
+Suppose you have two `m-by-n` matrix `X` and `Y`, then you can compute all
+distances between corresponding columns of `X` and `Y` in one batch, using
+the `colwise` function, as
 
 ```julia
 r = colwise(dist, X, Y)
 ```
 
-The output `r` is a vector of length `n`. In particular, `r[i]` is the distance between `X[:,i]` and `Y[:,i]`. The batch computation typically runs considerably faster than calling `evaluate` column-by-column.
+The output `r` is a vector of length `n`. In particular, `r[i]` is the distance
+between `X[:,i]` and `Y[:,i]`. The batch computation typically runs considerably
+faster than calling `evaluate` column-by-column.
+
+Note that either of `X` and `Y` can be just a single vector -- then the `colwise`
+function will compute the distance between this vector and each column of the other
+parameter.
+
+The above extends to iterators of iterators/vectors, and the colwise computations
+means the distance computation of corresponding objects (à la `Base.zip`).
+For instance, in Julia v1.1 and above, the colwise behavior shown above can be
+equally well obtained by
+
+```julia
+r = colwise(dist, eachcol(X), eachcol(Y))
+```
+
+Equally simple, the analogous "rowwise" distance computation can be obtained via
+
+```julia
+r = colwise(dist, eachrow(X), eachrow(Y))
+```
 
-Note that either of `X` and `Y` can be just a single vector -- then the `colwise` function will compute the distance between this vector and each column of the other parameter.
+The iterator-based approach is in fact the future-proof way of computing distances
+between corresponding data points of iterable collections `X` and `Y`. Exceptions
+are given by distance types with `colwise` methods specialized for
+`AbstractVecOrMat` arguments, i.e., `CorrDist` and `[Sq]Mahalanobis`.
```

Comment:
The wording makes it sound like for these distances the iterator-based approach wouldn't be future-proof. But the problem is rather than they are currently slower. Same below for `pairwise`.